File: ui/src/main/java/io/github/aosn/camp2016/ui/logic/JsonSerializer.java
package io.github.aosn.camp2016.ui.logic;

import com.fasterxml.jackson.databind.ObjectMapper;
import io.github.aosn.util.tryable.Try;

public class JsonSerializer {
    private static final ObjectMapper MAPPER = new ObjectMapper();

    public static Try<String> serialize(Object o) {
        return Try.to(() -> MAPPER.writeValueAsString(o));
    }

    public static <T> Try<T> deserialize(String json, Class<T> dto) {
        return Try.to(() -> MAPPER.readValue(json, dto));
    }

    private JsonSerializer() {
    }
}

File: backend-play-scala/app/util/JsonSerializer.scala
package util

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.scala.DefaultScalaModule
import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper

import scala.util.Try

object JsonSerializer {
  private[this] val MAPPER = new ObjectMapper with ScalaObjectMapper
  MAPPER.registerModule(DefaultScalaModule)

  def serialize(o: AnyRef): String = {
    MAPPER.writeValueAsString(o)
  }

  def deserialize[T](json: String): Try[T] = {
    Try(MAPPER.readValue(json))
  }
}
